columnflow/tasks/framework/base.py

```diff
@@ -1434,96 +1459,95 @@ def resolve_shifts(cls, params: dict[str, Any]) -> dict:
                     params["global_shift_insts"] = {config_inst: params["global_shift_inst"]}
                     params["local_shift_insts"] = {config_inst: params["local_shift_inst"]}
 
-        else:
-            if (config_insts := params.get("config_insts")):
-                configs_repr = lambda: ", ".join(map(repr, config_insts))
-                empty_pair = (law.NO_STR, law.NO_STR)
-                nominal_pair = ("nominal", "nominal")
-                # evaluate shifts per config
-                shifts = {
-                    config_inst: cls._resolve_shifts(config_inst, params)
-                    for config_inst in config_insts
-                }
-                unique_shifts = set(shifts.values())
-                # when one shift pair is empty, all must be empty
-                if len(unique_shifts) > 1 and empty_pair in unique_shifts:
-                    raise ValueError(
-                        f"found invalid combination of resolved shifts {unique_shifts} in configs {configs_repr()}",
-                    )
-                # at least one shift pair must be known
-                if unique_shifts == {None}:
-                    raise ValueError(f"shift {params['shift']} unknown to configs {configs_repr()}")
-                # fill unknown shifts with the nominal one
-                if None in unique_shifts:
-                    shifts = {
-                        config_inst: _shifts or nominal_pair
-                        for config_inst, _shifts in shifts.items()
-                    }
-                    unique_shifts = set(shifts.values())
-
-                # determine the overall shift pair for this multi-config task and store parameters
-                non_nominal_shifts = unique_shifts - {nominal_pair}
-                if len(non_nominal_shifts) > 1:
+        elif (config_insts := params.get("config_insts")):
+            # a shift validatio task supporting a single config must be set
+            if cls.shift_validation_task_cls is None:
+                raise AttributeError(
+                    f"shift resolution of multi-config task '{cls}' requires 'shift_validation_task_cls' to be set",
+                )
+            if not cls.shift_validation_task_cls.has_single_config():
+                raise ValueError(
+                    f"shift_validation_task_cls '{cls.shift_validation_task_cls}' must have a single config",
+                )
```

Comment:
Code duplication (this is already part of `cls._resolve_upstream_shifts`
```suggestion
```

Reply:
True, I forgot I moved it.